module/documents/chat-message.mjs

```diff
@@ -738,6 +740,25 @@ export default class ChatMessage5e extends ChatMessage {
     }
   }
 
+  static activateListeners() {
+    document.addEventListener("keydown", event => {
+      if (event.key !== "Shift") return;
+      let deleteButtons = document.querySelectorAll(".chat-message .message-delete");
+      if (!deleteButtons.length) return;
+      if (deleteButtons[0].style.display === "none") {
+        deleteButtons.forEach(btn => btn.style.display = "");
+      }
+    });
+    document.addEventListener("keyup", event => {
+      if (event.key !== "Shift") return;
+      let deleteButtons = document.querySelectorAll(".chat-message .message-delete");
+      if (!deleteButtons.length) return;
+      if (deleteButtons[0].style.display === "") {
+        deleteButtons.forEach(btn => btn.style.display = "none");
+      }
+    });
```

Comment:
Looks good, and I had some suggestions:

1. We should probably make these listeners passive given that they never call `preventDefault`, (i.e. `document.addEventListener("keydown", event => { /* [snip] */ }, { passive: true });`
2. We probably need to add a `blur` listener on the window for edge cases where the user holds shift, clicks outside the window, then releases shift.
3. We could consider using the `hidden` attribute rather than setting `display: none`.
4. If we do use `hidden`, we could use `toggleAttribute`, which allows us to refactor this to share code a bit.

```js
/** JSDoc */
static activateListeners() {
  document.addEventListener("keydown", event => {
    if ( event.key === "Shift" ) this.toggleDeleteButtons(false);
  }, { passive: true });
  document.addEventListener("keyup", event => {
    if ( event.key === "Shift" ) this.toggleDeleteButtons(true);
  }, { passive: true });
  window.addEventListener("blur", () => this.toggleDeleteButtons(true), { passive: true });
}

/** JSDoc */
static toggleDeleteButtons(hide) {
  const deleteButtons = document.querySelectorAll(".chat-message .message-delete");
  deleteButtons.forEach(btn => btn.toggleAttribute("hidden", hide));
}
```

Reply:
Implemented these changes & added the missing JSDoc. Also switched earlier code in `_enrichChatCard` to use `hidden` instead of `display: none`.